google/cloud/storage/client.h

```diff
@@ -533,6 +533,40 @@ class Client {
     return raw_client_->PatchObject(request).second;
   }
 
+  /**
+   * Compose existing objects into a new object in the same bucket.
+   *
+   * @param bucket_name the name of the bucket that contains the object.
+   * @param destination_object_name the composed object name.
+   * @param source_objects objects used to compose `destination_object_name`.
+   * @param destination_object_metadata the new metadata for the Object. Only
+   *     the writeable fields accepted by the `Objects: compose` API are used,
+   *     all other fields are ignored.
+   * @param options a list of optional query parameters and/or request headers.
+   *     Valid types for this operation include `ComposeObjectRequest`,
+   *      `EncryptionKey`, `Generation`, `IfGenerationMatch`,
+   *      `IfMetagenerationMatch`, `PredefinedAcl`,
+   *      `SourceEncryptionKey`, `UserProject`.
+   *
+   * @throw std::runtime_error if the operation fails.
+   *
+   * @par Example
+   * @snippet storage_object_samples.cc update object metadata
+   */
+  template <typename... Options>
+  ObjectMetadata ComposeObject(std::string bucket_name,
+                        std::string destination_object_name,
```

Comment:
I think the indentation is off, but ignore what I think and just do whatever the build with `clang-format(1)` tells you.